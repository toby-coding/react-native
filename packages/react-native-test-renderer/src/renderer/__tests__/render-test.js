/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow
 */

'use strict';

import * as ReactNativeTestRenderer from '../index';
import * as React from 'react';
import {Text, View} from 'react-native';
import 'react-native/Libraries/Components/View/ViewNativeComponent';

function TestComponent() {
  return (
    <View>
      <Text>Hello</Text>
      <View />
    </View>
  );
}

function TestComponentWithProps() {
  return (
    <View pointerEvents="box-none">
      <Text>Hello</Text>
      <View style={{flex: 1}} />
    </View>
  );
}

describe('render', () => {
  describe('toJSON', () => {
    it('returns expected JSON output based on renderer component', () => {
      const result = ReactNativeTestRenderer.render(<TestComponent />);
      expect(result.toJSON()).toMatchSnapshot();
    });

    it('renders View props', () => {
      const result = ReactNativeTestRenderer.render(<TestComponentWithProps />);
      expect(result.toJSON()).toMatchSnapshot();
    });
  });

  describe('findAll', () => {
    it('returns all nodes matching the predicate', () => {
      const result = ReactNativeTestRenderer.render(<TestComponent />);
      const textNode = result.findAll(node => {
        return node.props?.text === 'Hello';
      })[0];
      expect(textNode).not.toBeUndefined();

      const viewNodes = result.findAll(node => {
        return node.viewName === 'RCTView';
      });
      expect(viewNodes.length).toBe(2);
    });
  });
});
