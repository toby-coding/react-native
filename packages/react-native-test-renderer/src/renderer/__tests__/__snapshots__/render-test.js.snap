// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render toJSON renders View props 1`] = `
<RCTView
  pointerEvents="box-none"
>
  <RCTText
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isHighlighted={false}
    selectionColor={null}
  >
    Hello
  </RCTText>
  <RCTView
    style={
      {
        "flex": 1,
      }
    }
  />
</RCTView>
`;

exports[`render toJSON returns expected JSON output based on renderer component 1`] = `
<RCTView>
  <RCTText
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isHighlighted={false}
    selectionColor={null}
  >
    Hello
  </RCTText>
  <RCTView />
</RCTView>
`;
